### PR TITLE
Release workflow: fix bumping version

### LINKED ISSUE #192: Test git tag push release workflow
Confirm that the push of a git tag formatted as `vX.XXX*` prepares a GitHub release with the appropriate release version, and a zip archive with the release files that includes the release version in the zip archive file path.

### PR DESCRIPTION
**Pull request by RickyDaMa**
Will close #192 

It's a Python script for bumping the font version. The old bash / in workflow version was unreliable (venv wasn't yet necessarily made) and pulled in a heavy dependecy (`yq`, a snap). This simple script aims to replace both

I've also pinned the Python version of the CI to 3.11 both for consistency and so we don't get thrust into 3.12 without warning, as Nikolaus is concerned with dependencies when that happens

Technical jargon:

I have to `subprocess.run` `bumpfontversion` because it's not designed to be run in a script as far as I can tell

~~Aside: `bumpfontversion` has no license, and is therefore proprietary (though [source available](https://github.com/simoncozens/bumpfontversion))~~ (edit: it's in the README of the repo)

### CODE REVIEW

File: scripts/bump-to-tag.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        file_text = CONFIG_FILE.read_text()
+    except (KeyError, FileNotFoundError):
+        print("This script is intended to be run by CI during a release. Exiting")
+        exit(2)
+
+    gftools_config = yaml.safe_load(file_text)
+    for source in gftools_config["sources"]:
+        print(f"Bumping {source} to {tag_name}")
+        cmd = subprocess.run(
+            f". venv/bin/activate && bumpfontversion sources/{source} --new-version {tag_name}",
```
Comment (madig) [inline scripts/bump-to-tag.py]: I think the activation here may not be neccessary if it's done in the Makefile anyway?

Comment (RickyDaMa) [reply scripts/bump-to-tag.py]: I will test this in playground, though I suspect you're right

Comment (RickyDaMa) [reply scripts/bump-to-tag.py]: Activation was not necessary, but `shell=True` still is

**Comment by RickyDaMa** [comment]
@MariannaPaszkowska this is ready to merge now. Means that creating a release on GitHub will automatically attach the font archive to the release

**Pull request merged**